code/modules/mob/living/silicon/robot/syndicate_robot.dm

```diff
@@ -68,6 +73,23 @@
 		return
 	src.ckey = user.ckey
 	SSghostroles.remove_spawn_atom("syndiborg", src)
+	// uggo but i think this is the best possible way of doing this
+	if(!assigned_antagonist)
+		if(traitors)
+			assigned_antagonist = traitors
+		else if(mercs)
+			assigned_antagonist = mercs
+		else if(raiders)
+			assigned_antagonist = raiders
+		else if(revs)
+			assigned_antagonist = revs
+		else if(ninjas)
+			assigned_antagonist = ninjas
+	if(assigned_antagonist)
+		assigned_antagonist.add_antagonist_mind(src.mind, TRUE)
+		if(assigned_antagonist.get_antag_radio())
+			module.channels[assigned_antagonist.get_antag_radio()] = TRUE
+			radio.recalculateChannels()
```

Comment:
This seems like a weird way to handle the radio channels.

Wouldnt it be possible to just set a variable based on the antag type of the person that activates the spawner to decide which antag channel they get?

Reply:
it already does that, this is just an emergency fallback for if someone that doesn't have an antag set activates it

Reply:
Then it should imho just be nixed.
The borg spawner is only available to antags unless there is admin intervention, and if there is admin intervention, they can also take care of the radio channels.